app/lib/foreman_deployments/tasks/wait_until_built_task_definition.rb

```diff
@@ -0,0 +1,60 @@
+module ForemanDeployments
+  module Tasks
+    class WaitUntilBuiltTaskDefinition < ForemanDeployments::Tasks::BaseDefinition
+      class Action < BaseAction
+        include Dynflow::Action::Polling
+
+        def done?
+          external_task['built'] == true
+        end
+
+        def timeout
+          input['timeout'] || 2 * 60 * 60 # 2 hours default
+        end
+
+        def invoke_external_task
+          schedule_timeout(timeout) unless timeout <= 0
+          WaitUntilBuiltTaskDefinition.build_status
+        end
+
+        def poll_external_task
+          # TODO: check the case when no host_id is sent
```

Comment:
Done. I added the same check to the dynflow action itself too, to catch the cases when nil would be propagated due to a failure in a previous task.